### PR TITLE
Add chapter on software versioning

### PR DESCRIPTION
**Pull request by ajaust**
## Description

This PR adds slides for the software versioning chapter.

## Checklist

- [x] I made sure that the markdown files are formatted properly.
- [x] I made sure that all hyperlinks are working.
- [ ] I added the new content to the `README.md` inside the topics subdirectory, e.g., `01_version_control/README.md`.
- [ ] I added the new content to the `timetable.md` in the root of this repository.
- [ ] I make sure to update the content/links on the [homepage](https://github.com/Simulation-Software-Engineering/homepage).

### CODE REVIEW

**Comment by ajaust** [comment]
@uekerman Should we define learning goals for the misc slides? I did not any at the moment as I had the feeling it might get very fine-grained.

**Comment by uekerman** [comment]
> @uekerman Should we define learning goals for the misc slides? I did not any at the moment as I had the feeling it might get very fine-grained.

Thought about the same. Not sure if we have common learning goals for all of these things. It is more a

> There important things that we have not yet covered and that you should have seen at least once.

We could make this our learning goal in the chapter README for the moment.

File: 06_miscellaneous/versioning_slides.md [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+
+---
+
+## Semantic Versioning
+
+```text
+MAJOR.MINOR.PATCH-<pre-release>+<build>
+```
+
+- `MAJOR` version when you make incompatible API changes,
```
Comment (uekerman) [inline 06_miscellaneous/versioning_slides.md:79]: No need to write this down necessarily, but I guess we might need to explain here a bit:

a) what incompatible changes are (could also be more than API, e.g. configuration files)
b) why they are bad -> everything that depends on you needs to update

**Review by uekerman — APPROVED** [review]
Looking good.

**Pull request merged**